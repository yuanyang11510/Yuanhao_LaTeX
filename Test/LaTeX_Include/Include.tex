\documentclass{article}
\usepackage{import} % import宏包提供了"\import{full path}{file}"命令和"\subimport{relative path}{file}"命令，前者类似"\input{}"命令，后者可以嵌套使用：在文件一中使用"\import{full path}{file}"命令引用文件二，在文件二中使用"\subimport{relative path}{file}"命令引用文件三，这样就可以直接在文件一中使用"\import{full path}{file}"命令引用文件三
%rfr 可以参考import宏包的官方手册：https://ftp.kddilabs.jp/CTAN/macros/latex/contrib/import/import.pdf，其中关于相对路径的说明："The 〈full-path〉 argument for \import can be a full absolute path or a relative path starting from the main working directory for the document."

\input{../LaTeX_Include/Package/Package.tex}
% \import{../LaTeX_Include/Package/}{Package.tex}

%! "\include{}"命令会使编译报错
% \include{../LaTeX_Include/Package/Package.tex}
% \includeonly{Chapter_1.tex,Chapter_2.tex}
%rfr 这篇帖子（https://cfranc.wordpress.com/2009/12/01/include-input-and-relative-paths-in-latex/）当中有答案认为是MiKTeX的bug，但是经检验，即使安装了TeXLive并卸载了MiKTeX，问题仍然存在

\begin{document}

% "\import"命令
\import{../LaTeX_Include/Chapter1/}{Chapter_1.tex}
%* 嵌套命令可以使用"\import-\input"，也可以使用"\import-\subimport"，但是不能使用"\input-\input"，也不能够使用"\import-\import"
% \import{../LaTeX_Include/Chapter2/}{Chapter_2.tex}
% \import{../LaTeX_Include/Forbidden/}{Forbidden.tex}

% "\input"命令
%! \input{../LaTeX_Include/Chapter1/Chapter_1.tex}
%! 不要使用上面这条命令，因为引用的文件"Chapter_1.tex"中存在"\input{}"命令
\input{../LaTeX_Include/Chapter2/Chapter_2.tex}
\input{../LaTeX_Include/Forbidden/Forbidden.tex}

% "\include"命令
% "\include"命令会在引用文件前添加"\clearpage"命令进行分页
%! "\include{}"命令会使编译报错
% \include{../LaTeX_Include/Chapter1/Chapter_1.tex}
% \include{../LaTeX_Include/Chapter2/Chapter_2.tex}
% \include{../LaTeX_Include/Forbidden/Forbidden.tex}

\end{document}