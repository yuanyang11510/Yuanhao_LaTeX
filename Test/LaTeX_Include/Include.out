\BOOKMARK [1][-]{section.1}{\376\377\116\000}{}% 1
\BOOKMARK [2][-]{subsection.1.1}{\376\377\377\010\116\000\377\011}{section.1}% 2
\BOOKMARK [2][-]{subsection.1.2}{\376\377\116\214}{section.1}% 3
\BOOKMARK [1][-]{section.2}{\376\377\116\214}{}% 4
\BOOKMARK [1][-]{section.3}{\376\377\171\201}{}% 5
